### PR TITLE
Severial fixes and improvements on the Makefile and the script

### PR DESCRIPTION
**Pull request by cyyself**
Several fixes and improvements on the Makefile and the script, including:
- Makefile: Do not override SPEC in Makefile
- README: add step to init data
- collect.sh: do not collect log files
- README: fix usage of collect data
- Makefile: introduce new OPTIMIZE flag
- README: clean up

The detailed information is in each commit message.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,10 @@ make && make install
 // prepare jemalloc compiled before
 export RISCV=/riscv_toolchain/top
 export LD_JEMALLOC=1
-make build-all -j 70
+make ARCH=riscv64 \
+     CROSS_COMPILE=riscv64-unknown-linux-gnu- \
+     OPTIMIZE="-O3 -flto -march=rv64gcv_zba_zbb_zbc_zbs -mtune=xiangshan-nanhu" \
```
Comment (shinezyy) [inline README.md:70]: mtune 只有特定版本的gcc 才有吧？

File: README.md [hunk 2]
```diff
@@ -23,29 +17,35 @@ export SPEC=/spec2006_path
...
+bash scripts/src_copy.sh
 ```
 - compile binarys
 ```
-export ARCH=riscv64
-export CROSS_COMPILE=riscv64-unknown-linux-gnu-
-make build-all -j 70
+make ARCH=riscv64 \
+     CROSS_COMPILE=riscv64-unknown-linux-gnu- \
+     OPTIMIZE="-O3 -flto -march=rv64gcv_zba_zbb_zbc_zbs -mtune=xiangshan-nanhu" \
```
Comment (shinezyy) [inline README.md:35]: mtune=xiangshan-nanhu 只有特定版本的gcc 才有吧？

**Review by shinezyy — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (cyyself) [reply README.md:35]: Yes. At least GCC 14.1. Just place here to let users know we can use Xiangshan parameters for better performance.

Comment (shinezyy) [reply README.md:35]: > Yes. At least GCC 14.1. Just place here to let users know we can use Xiangshan parameters for better performance.

You can put it in section Note for GCC >= 14

**Review by shinezyy — APPROVED** [review]

**Pull request merged**